compiler/ilgen/IlBuilder.cpp

```diff
@@ -1380,24 +1392,59 @@ IlBuilder::EqualTo(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+void
+IlBuilder::integerizeAddresses(TR::IlValue **leftPtr, TR::IlValue **rightPtr)
+   {
+   TR::IlValue *left = *leftPtr;
+   TR::IlValue *right = *rightPtr;
+   if (left->getSymbol()->getDataType() == TR::Address)
+      {
+      TR_ASSERT(right->getSymbol()->getDataType() == TR::Address, "mismatched types in address compare");
+      *leftPtr = ConvertTo(Word, left);
+      *rightPtr = ConvertTo(Word, right);
+      }
+   }
+
 TR::IlValue *
 IlBuilder::LessThan(TR::IlValue *left, TR::IlValue *right)
    {
+   integerizeAddresses(&left, &right);
    TR::IlValue *returnValue=compareOp(TR_cmpLT, false, left, right);
    TraceIL("IlBuilder[ %p ]::%d is LessThan %d < %d?\n", this, returnValue->getCPIndex(), left->getCPIndex(), right->getCPIndex());
    ILB_REPLAY("%s = %s->LessThan(%s, %s);", REPLAY_VALUE(returnValue), REPLAY_BUILDER(this), REPLAY_VALUE(left), REPLAY_VALUE(right));
    return returnValue;
    }
 
 TR::IlValue *
+IlBuilder::LessOrEqualTo(TR::IlValue *left, TR::IlValue *right)
+   {
+   integerizeAddresses(&left, &right);
+   TR::IlValue *returnValue=compareOp(TR_cmpLE, false, left, right);
+   TraceIL("IlBuilder[ %p ]::%d is LessOrEqualTo %d < %d?\n", this, returnValue->getCPIndex(), left->getCPIndex(), right->getCPIndex());
```

Comment:
Should the message use `<=` instead of just `<` or is this symbol reserved for something else?